packages/checkout-storefront/src/hooks/useFormDebouncedSubmit.ts

```diff
@@ -0,0 +1,66 @@
+import { FormDataBase } from "@/checkout-storefront/lib/globalTypes";
+import { debounce, isEqual, throttle } from "lodash-es";
+import { useCallback, useEffect, useRef } from "react";
+import { Path, UseFormReturn } from "react-hook-form";
+
+interface UseFormAutofillSubmit<TFormData extends FormDataBase>
+  extends Pick<UseFormReturn<TFormData>, "trigger" | "formState"> {
+  onSubmit: (formData: TFormData) => Promise<void> | void;
+  formData: TFormData;
+  defaultFormData?: TFormData;
+  autoSave?: boolean;
+}
+
+export const useFormDebouncedSubmit = <TFormData extends FormDataBase>({
+  formData,
+  defaultFormData,
+  onSubmit,
+  trigger,
+  autoSave = true,
+  formState,
+}: UseFormAutofillSubmit<TFormData>) => {
+  const { isDirty, dirtyFields } = formState;
+  const formDataRef = useRef<TFormData | undefined>(defaultFormData);
+
+  const debouncedSubmit = useCallback(
+    debounce((formData: TFormData) => {
+      if (!isDirty) {
```

Comment:
We might get stale value here. `isDirty` is the value of `isDirty` from line `22` but 2 seconds in the past. This might or might not be a problem. Not sure.
Implementing correct `useDebouncedCallback` or `useThrottledCallback` is not quite as simple.